d2core/d2asset/animation_manager.go

```diff
@@ -20,10 +20,27 @@ func createAnimationManager() *animationManager {
 	return &animationManager{d2common.CreateCache(animationBudget)}
 }
 
+// Dispose of all the animations to prevent ebiten storing the images
+func (am *animationManager) ClearCache() {
+	keys := am.cache.GetKeys()
+	for idx := range keys {
+		key := keys[idx]
```

Comment:
this is making a local copy of `keys[idx]`, i think it should be `key := &keys[idx]`